src/file_pipe_log/pipe.rs

```diff
@@ -254,6 +254,13 @@ impl<F: FileSystem> SinglePipe<F> {
         reader.read(handle)
     }
 
+    fn async_read(&self, blocks: Vec<FileBlockHandle>, ctx: &mut F::AsyncIoContext) {
```

Comment:
Move ctx first, as explained in https://github.com/tikv/raft-engine/pull/286#discussion_r1118813985.